pkg/integrations/v4/supervisor_fb_test.go

```diff
@@ -105,3 +106,87 @@ func TestFBSupervisorConfig_LicenseKeyShouldBePassedAsEnvVar(t *testing.T) {
 	assert.Contains(t, exec.(*executor2.Executor).Cfg.Environment, "NR_LICENSE_KEY_ENV_VAR")       // nolint:forcetypeassert
 	assert.Equal(t, exec.(*executor2.Executor).Cfg.Environment["NR_LICENSE_KEY_ENV_VAR"], license) //nolint:forcetypeassert
 }
+
+func TestRemoveFbConfigTempFiles(t *testing.T) {
+
+	configFiles := []struct {
+		name    string
+		content string
+	}{
+		{"nr_fb_config1", "nr_fb_lua_filter1,nr_fb_lua_filter2"},
+		{"nr_fb_config2", ""},
+		{"nr_fb_config3", "nr_fb_lua_filter3"},
+		{"nr_fb_config4", "nr_fb_lua_filter4"},
+		{"nr_fb_config5", ""},
+		{"nr_fb_config6", ""},
+		{"nr_fb_lua_filter1", ""},
+		{"nr_fb_lua_filter2", ""},
+		{"nr_fb_lua_filter3", ""},
+		{"nr_fb_lua_filter4", ""},
+	}
+
+	tests := []struct {
+		name                     string
+		maxNumConfFiles          int
+		expectedRemovedConfFiles []string
+		expectedKeptConfFiles    []string
+		wantErr                  bool
+	}{
+		{
+			name:                     "No config files are removed",
+			maxNumConfFiles:          10,
+			expectedRemovedConfFiles: []string{},
+			expectedKeptConfFiles:    []string{"nr_fb_config1", "nr_fb_config2", "nr_fb_config3", "nr_fb_config4", "nr_fb_config5", "nr_fb_config6", "nr_fb_lua_filter1", "nr_fb_lua_filter2", "nr_fb_lua_filter3", "nr_fb_lua_filter4"},
+			wantErr:                  false,
+		},
+		{
+			name:                     "Config file 1 and config lua files 1 and 2 are removed",
+			maxNumConfFiles:          5,
+			expectedRemovedConfFiles: []string{"nr_fb_config1", "nr_fb_lua_filter1", "nr_fb_lua_filter2"},
+			expectedKeptConfFiles:    []string{"nr_fb_config2", "nr_fb_config3", "nr_fb_config4", "nr_fb_config5", "nr_fb_config6", "nr_fb_lua_filter3", "nr_fb_lua_filter4"},
+			wantErr:                  false,
+		},
+		{
+			name:                     "Config files 1, 2 and 3 and config lua files 1, 2 and 3 are removed",
+			maxNumConfFiles:          3,
+			expectedRemovedConfFiles: []string{"nr_fb_config1", "nr_fb_config2", "nr_fb_config3", "nr_fb_lua_filter1", "nr_fb_lua_filter2", "nr_fb_lua_filter3"},
+			expectedKeptConfFiles:    []string{"nr_fb_config4", "nr_fb_config5", "nr_fb_config6", "nr_fb_lua_filter4"},
+			wantErr:                  false,
+		},
```

Comment:
Can you please add a test that attempts to remove a config file that points to a non-existing lua script in the middle of the file iteration? This one should be successful in removing the initial files in the list, but fail at removing this one with the broken link to the LUA file. The test should expect a partially-filled list of `expectedRemovedFiles`, and `wantErr: true`.

Reply:
I've tested that possibility manually, but I'll create an specific unit test